helix-core/src/test/java/org/apache/helix/integration/task/TestNoDoubleAssign.java

```diff
@@ -42,7 +42,7 @@
 import org.testng.annotations.Test;
 
 public class TestNoDoubleAssign extends TaskTestBase {
-  private static final int THREAD_COUNT = 10;
+  private static final int THREAD_COUNT = 1;
```

Comment:
Since thread count is 1, I think `newSingleThreadExecutor` is what you need?

Reply:
changed to use newSingleThreadExecutor for break connection